src/server/accounting.c

```diff
@@ -1147,56 +1147,60 @@ account_jobend(job *pjob, char *used, int type)
 	len -= i;
 
 	/* finally add on resources used from req_jobobit() */
-	if (((used == NULL) && (pjob->ji_acctrec == NULL) && (type == PBS_ACCT_END)) ||
-		!(used && strstr(used, "resources_used") && (type == PBS_ACCT_RERUN))) {
-		/* If pbs_server is restarted during the end of job processing then used maybe NULL.
-		 * So we try to derive the resource usage information from resources_used attribute of
-		 * the job and then reconstruct the resources usage information into resc_used buffer.
-		 */
-		if (pjob->ji_wattr[(int)JOB_ATR_resc_used].at_user_encoded != NULL)
-			patlist = pjob->ji_wattr[(int)JOB_ATR_resc_used].at_user_encoded;
-		else if (pjob->ji_wattr[(int)JOB_ATR_resc_used].at_priv_encoded != NULL)
-			patlist = pjob->ji_wattr[(int)JOB_ATR_resc_used].at_priv_encoded;
-		else
-			encode_resc(&pjob->ji_wattr[(int)JOB_ATR_resc_used],
-				&temp_head, job_attr_def[(int)JOB_ATR_resc_used].at_name,
-				(char *)0, ATR_ENCODE_CLIENT, &patlist);
+	if (type == PBS_ACCT_END || type == PBS_ACCT_RERUN) {
+		if ((used == NULL && pjob->ji_acctrec == NULL) || (used != NULL && strstr(used, "resources_used") == NULL)) {
```

Comment:
if you run the PTL test cases from pbs_Rrecord_resources_used.py, test_t7 will fail with this fix. This part needs to be reworked..